core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -223,9 +225,13 @@ public void testToQuery() throws IOException {
 
         Query expectedQuery = createExpectedQuery(testQuery, context);
         Query actualQuery = testQuery.toQuery(context);
-        assertThat(actualQuery, instanceOf(expectedQuery.getClass()));
-        assertThat(actualQuery, equalTo(expectedQuery));
-        assertLuceneQuery(testQuery, actualQuery, context);
+        if (expectedQuery == null) {
```

Comment:
I also need this already in #11915 when I add possibility of tests with inner query builders that return null (EmptyQueryBuilder).